core/arch/arm/mm/core_mmu.c

```diff
@@ -988,6 +988,78 @@ static void set_pg_region(struct core_mmu_table_info *dir_info,
 	}
 }
 
+TEE_Result core_mmu_map_pages(vaddr_t start, paddr_t *pages, size_t num_pages,
+			      enum teecore_memtypes memtype)
+{
+	TEE_Result ret;
+	struct core_mmu_table_info tbl_info;
+	unsigned int idx;
+	uint32_t old_attr;
+	vaddr_t vaddr = start;
+	size_t i;
+
+	if (vaddr & SMALL_PAGE_MASK)
+		return TEE_ERROR_BAD_PARAMETERS;
```

Comment:
Yes, restricting memtype to `MEM_AREA_SHM_VASPACE` is ok. This will rely on the counterpart that guaranties the physical base addresses of registered SHM. I think we could open this `core_mmu_map_page()` to SEC_RAM and NSEC_RAM and SDP_MEM.

Mapping to `MEM_AREA_RES_VASPACE` will lead nowhere as the relate memory attribute in not valid (bit0!=1). We should update `init_fdt()` to use this `core_mmu_[un]map_pages()` at dtb parsing and remove this `MEM_AREA_RES_VASPACE`. but that another story.